src/renderers/dom/shared/__tests__/ReactDOMServerIntegration-test.js

```diff
@@ -347,6 +363,91 @@ describe('ReactDOMServerIntegration', () => {
       expect(e.childNodes.length).toBe(1);
       expect(e.firstChild.tagName).toBe('BR');
     });
+
+    if (ReactDOMFeatureFlags.useFiber) {
+      itRenders('a array type children as a child', async render => {
+        let Header = props => {
+          return <p>header</p>;
```

Comment:
Let's also make the component return array as well. This way we verify it's supported at multiple levels rather than just top level.